test/gtest/nonpack_conv3d_fwd.hpp

```diff
@@ -0,0 +1,125 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2023 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+#pragma once
+
+#include "conv3d_test_case.hpp"
+
+struct NonPackTestCase : Conv3DTestCase
+{
+    size_t s0;
+    size_t s1;
+    size_t s2;
+    size_t s3;
+    size_t s4;
+    std::vector<size_t> GetInputStrides() { return {s0, s1, s2, s3, s4}; }
```

Comment:
@iq136boy : to be more thorough, you should also pass strides for `Weight` and `Output` tensors. One way to generate correct strides is to multiply `packed` strides with a small constant. E.g., you have packed dims as `{N, C, D, H, W}`, you can produce non-packed strides as `{2 * C* D* H * W, 2 * D * H * W, 2 * H * W, 2 * W, 2 * 1}`. 

You can also take it one step further and multiply a prefix of strides by a constant, e.g., (for the case above):
`{2 * C* D* H * W, 2 * D * H * W, 2 * H * W,  W, 1}` is a valid non-packed strides value where only a prefix of strides are non-packed and remainder is packed.

Reply:
@amberhassaan I agreed that we should also put the strides for weight and output. I also agree that there are more ways that the non-packed tensors can be generated. However, the user should give the strides of nonpacked tensors anyway. What we do need is just to check if the tensor strides are valid or not.

Reply:
Yes, but these are tests you are writing and we need to test if convolution works for non-packed inputs. If we don't test a feature then we can't offer a guarantee that it works.

Reply:
> Yes, but these are tests you are writing and we need to test if convolution works for non-packed inputs. If we don't test a feature then we can't offer a guarantee that it works.

Oh, I see. You mean add more test cases. That make sense.

Reply:
@iq136boy : please create an issue for this item. I can take it up. Basically, I want to generate non-packed strides for all 3 of input, output and weights. CC: @JehandadKhan